packages/cdktf/lib/terraform-element.ts

```diff
@@ -38,26 +46,51 @@ export class TerraformElement extends Construct {
     return {};
   }
 
+  private _fqnToken?: string;
+
+  public get fqn() {
+    if (!this._fqnToken) {
+      ok(!!this._elementType, "Element type not set");
```

Comment:
nice! I like this. We should use the `assert` package more often 😇

Reply:
We switched from if's to ok's, no regrets, it's easy to read and smaller at the same time, win-win! :)